GVFS/GVFS.UnitTests/Common/NuGetUpgrade/NuGetUpgraderTests.cs

```diff
@@ -54,20 +57,26 @@ public void SetUp()
                 this.downloadDirectoryPath,
                 null,
                 this.tracer);
+
             this.mockFileSystem = new MockFileSystem(
                 new MockDirectory(
                     Path.GetDirectoryName(this.downloadDirectoryPath),
                     new[] { new MockDirectory(this.downloadDirectoryPath, null, null) },
                     null));
 
+            this.mockCredentialManager = new Mock<ICredentialManager>();
+            string credentialManagerString = "value";
+            this.mockCredentialManager.Setup(foo => foo.TryGetCredential(It.IsAny<ITracer>(), It.IsAny<string>(), out credentialManagerString, out credentialManagerString, out credentialManagerString)).Returns(true);
```

Comment:
Seems like with this mock we could now test failure on the first `TryGetCredential` and make sure the delete and store get called and it is retried properly?

Reply:
Yep! see `QueryNewestVersionReacquiresCredentialsOnAuthFailure`